presto-main/src/main/java/com/facebook/presto/type/IpPrefixType.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.block.PageBuilderStatus;
+import com.facebook.presto.spi.block.VariableWidthBlockBuilder;
+import com.facebook.presto.spi.type.AbstractType;
+import com.facebook.presto.spi.type.FixedWidthType;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.google.common.net.InetAddresses;
+import io.airlift.slice.Slice;
+import io.airlift.slice.XxHash64;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+
+public class IpPrefixType
+        extends AbstractType
+        implements FixedWidthType
+{
+    public static final IpPrefixType IPPREFIX = new IpPrefixType();
+
+    private IpPrefixType()
+    {
+        super(parseTypeSignature(StandardTypes.IPPREFIX), Slice.class);
+    }
+
+    @Override
+    public int getFixedSize()
+    {
+        return SIZE_OF_LONG * 2 + 1;
+    }
+
+    @Override
+    public BlockBuilder createBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries, int expectedBytesPerEntry)
+    {
+        int maxBlockSizeInBytes;
+        if (blockBuilderStatus == null) {
+            maxBlockSizeInBytes = PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
+        }
+        else {
+            maxBlockSizeInBytes = blockBuilderStatus.getMaxPageSizeInBytes();
+        }
+        return new VariableWidthBlockBuilder(
```

Comment:
I don't like using `VariableWidthBlockBuilder` for this. Each entry should be 5 bytes, and it can be fixed width. I wonder why we don't have a generic `FixedWidthBlockBuilder`. @wenleix do you have any thoughts?